### PR TITLE
Update all projects dependencies

### PR DESCRIPTION
**Pull request by 0237h**
# Changes

## Project

- Remove `@babel/core`
- Remove `@babel/preset-env`
- Remove `@babel/preset-typescript`
- `eslint`
	- Fix [`@typescript-eslint/no-this-alias`](https://typescript-eslint.io/rules/no-this-alias/) lint rule
	- Fix [`unicorn/prefer-node-protocol`](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-node-protocol.md) lint rule (add `node:` prefix for built-in modules)
	- Move `.eslintignore` inside config with `ignorePatterns`
	- Update config annotations
	- Use `ESLINT_USE_FLAT_CONFIG=false` in `pnpm` scripts

## `cli`

- Add `node:` prefix for built-in modules (e.g. `node:fs`)
- Remove `dockerode`
- Remove `tern`
- Upgrade `moduleResolution` to `bundler`, `module` to `ESNext` and `target` to `ESNext` making it an ESM module
- `binary-install-raw`
	- Replace with `binary-install`
- `chokidar`
	- Import type `FSWatcher` as namespace no longer available
	- Remove `await` for `onTrigger` as it's no longer async
- `http-ipfs-client`
	- Migrate from `http-ipfs-client` to `kubo-rpc-client` with dynamic import as it's ESM only[^1]
	- Make `createCompiler` async due to dynamic import
- `immutable`
	- `Map([ [ 'k', 'v' ] ])` or `Map({ k: 'v' })` instead of `Map.of`[^2]
	- Replace `immutable.Collection<ProtocolName, string[]>` with `immutable.Collection<ProtocolName, immutable.List<string>>`
- `js-yaml`
	- Replace `safe*` variants with `load`/`loadall`/`dump`[^3]
- `oclif/core`
	- Use `gluegun.prompt` instead of `ux.prompt`[^4]
	- Fix `prompt` variable assignments from object destructuring (thanks @YaroShkvorets)
	- Migrate to ESM[^5]
- `sync-request` 
	- Replace with async `fetch` call[^6]
	- Make `generateTypes` async and update NEAR test snapshot
- `vitest`
	- Remove `concurrent` for tests using the file system
- `web3-eth-abi`
	- Import `decodeLogs` method directly as there is no more default export
- `yaml`
	- Remove `strOptions`[^7]

## `ts`

- Remove `common/eager_offset.ts`
- `assemblyscript/asc`
	- Use `assemblyscript/asc` as import
	- `asc.main` no longer as callback for error, uses stderr and is async[^8]
	- Remove `asc.ready`[^9]
	- Remove `ascMain` in `package.json`[^10]
	- Replace `Promise.allSettled` with for loop in tests for resolving WASM generation promises in sequence (prevent WASM file corruption since they use same file `outputWasmPath`).
	- Silence warning 235 on compilation about exports[^11]

## `website`

- Fix `eslint` config

## Examples

- `ethereum-basic-event-handlers`
	- Fix `chai` max version to `4.x.x` as `chai 5.x.x` is ESM and `hardhat` doesn't support ESM Typescript projects yet[^12]
	- Rename `ethers.BaseContract.address` to `ethers.BaseContract.target`
	- Remove `@nomiclabs/hardhat-ethers`
	- Remove `@nomiclabs/hardhat-etherscan`
	- Remove `@typechain/ethers-v5`
	- Remove `@typechain/hardhat`
	- Remove `ethers`
	- Remove `hardhat-gas-reporter`
	- Remove `solidity-coverage`
	- Remove `typechain`

[^1]: https://github.com/ipfs/helia/issues/157
[^2]: https://github.com/immutable-js/immutable-js/blob/b3a1c9f13880048d744366ae561c39a34b26190a/CHANGELOG.md#breaking-changes
[^3]: https://github.com/nodeca/js-yaml/blob/0d3ca7a27b03a6c974790a30a89e456007d62976/migrate_v3_to_v4.md#safeload-safeloadall-safedump--load-loadall-dump
[^4]: https://github.com/oclif/core/discussions/999
[^5]: https://oclif.io/docs/esm/
[^6]: https://www.npmjs.com/package/sync-request
[^7]: https://github.com/eemeli/yaml/pull/235
[^8]: https://github.com/AssemblyScript/assemblyscript/releases/tag/v0.20.0
[^9]: https://github.com/AssemblyScript/assemblyscript/releases/tag/v0.20.0
[^10]: https://github.com/AssemblyScript/assemblyscript/releases/tag/v0.20.0
[^11]: https://github.com/AssemblyScript/assemblyscript/issues/2277
[^12]: https://hardhat.org/hardhat-runner/docs/advanced/using-esm#esm-and-typescript-projects

### CODE REVIEW

File: .eslintrc.cjs [hunk 1]
```diff
@@ -1,5 +1,17 @@
...
+  ignorePatterns: [
+    'node_modules',
+    'dist',
+    'build',
+    'generated',
+    'packages/cli/tests/cli/init',
+    'packages/cli/tests/cli/validation',
+    'packages/ts/test/',
+    'examples',
+    'cf-pages/*',
```
Comment (YaroShkvorets) [inline .eslintrc.cjs:12]: `cf-pages` is no longer there, maybe others can be removed too?

Comment (0237h) [reply .eslintrc.cjs]: Removed :+1:

File: .eslintrc.cjs
[the remark this answers is not in the document]
Comment (0237h) [reply .eslintrc.cjs:1]: I went ahead and removed some of them that were an easy fix. Some will need more work that we should tackle in a separate PR perhaps.

**Review by YaroShkvorets — COMMENTED** [review]
Made a few changes:
- removed `packages/cli/yarn.lock` since we are using pnpm throughout
- removed babel.config.js
- migrated eslint config to latest standard
- reviewed eslint rules 
- fixed various lint warnings

**Review by YaroShkvorets — APPROVED** [review]

**Pull request merged**